app/src/main/java/jembi/org/nfcdemo/utils/NdefFormatterTask.java

```diff
@@ -0,0 +1,78 @@
+package jembi.org.nfcdemo.utils;
+
+import android.nfc.FormatException;
+import android.nfc.NdefMessage;
+import android.nfc.NdefRecord;
+import android.nfc.Tag;
+import android.nfc.tech.Ndef;
+import android.os.AsyncTask;
+
+import java.io.IOException;
+import java.io.UnsupportedEncodingException;
+
+/**
+ * Created by Jembi Health Systems on 2017/10/09.
+ */
+
+public class NdefFormatterTask extends AsyncTask<Void, Void, Void> {
+
+    private Tag myTag;
+    private NfcFormatterCallback callback;
+
+    public NdefFormatterTask(Tag myTag, NfcFormatterCallback callback)  {
+        this.myTag = myTag;
+        this.callback = callback;
+    }
+
+    @Override
+    protected Void doInBackground(Void... voids) {
+        Ndef tag = Ndef.get(myTag);
+        try {
+            connectTag(tag);
+            tag.writeNdefMessage(createRecord());
+        } catch (FormatException | IOException ex) {
```

Comment:
To be extra safe, I'd recommend catching any exception rather than specific exceptions. Otherwise if another exception is thrown (that isn't FormatException or IOException) it will be sent to the UncaughtExceptionHandler. It would be better to redirect to the appropriate callback method.